src/rust/runtime/scheduler/scheduler.rs

```diff
@@ -43,31 +40,24 @@ use ::std::{
     },
 };
 
-//======================================================================================================================
-// Constants
-//======================================================================================================================
-
-/// Seed for the random number generator used to generate tokens.
-/// This value was chosen arbitrarily.
-#[cfg(debug_assertions)]
-const SCHEDULER_SEED: u64 = 42;
-const MAX_NUM_TASKS: usize = 16000;
-const MAX_RETRIES_TASK_ID_ALLOC: usize = 500;
-
 //======================================================================================================================
 // Structures
 //======================================================================================================================
 
+#[derive(Clone, Copy, Hash, PartialEq, Eq, Debug)]
+pub struct TaskId(pub u64);
+#[derive(Clone, Copy, Debug)]
+struct InternalId(usize);
+
 /// Task Scheduler
 pub struct Scheduler {
+    task_ids: IdMap<TaskId, InternalId>,
     /// Stores all the tasks that are held by the scheduler.
     tasks: PinSlab<Box<dyn Task>>,
-    /// Maps between externally meaningful ids and the index of the task in the slab.
-    task_ids: HashMap<u64, usize>,
     /// Holds the waker bits for controlling task scheduling.
     waker_page_refs: Vec<WakerPageRef>,
-    /// Small random number generator for tokens.
-    rng: SmallRng,
+    /// Holds the total number of running tasks.
+    total_num_tasks: usize,
```

Comment:
What is the purpose of this variable? I see that it is used on test code only.

Reply:
Sure, let me take it out of here because it is only needed for direct mapping.